### ISSUE TITLE
Error when I try to add an new integration

### ISSUE THREAD

**Issue by WichtP** [websylv/homeassistant-meteoswiss#74]
If I try to add a new integration I see and I can choose the meteo suisse integration but I always receive this error message:

Erreur
Le flux de configuration n'a pas pu être chargé: undefined

OK
 Can you help me please. Best regards 

PA

**Comment by albertogmdp** [comment]
I have exactly the same error. This is a fresh installation. I manage to download it from hacks but when trying to add the integration it gives the "flux-underfined" error.

**Comment by WichtP** [comment]
Thanks for you information. I don't know if anybody can help us to solve this error. Best regards. PA

**Comment by albertogmdp** [comment]
@websylv is the one. Are we missing something in our installations? Is this a known issue?

By the way, thank you for developing this integration. We have a great Meteo service in Switzerland so it is great to have it in home assistant.

**Comment by WichtP** [comment]
I agree with you Alberto, it's great that we have people to develop this kind of service. Best regards. PA

**Comment by rodonile** [comment]
I'm also having this error. Looking at the logs it looks like a python issue in the code:
`KeyError: 'Altitude station m. s. mer'`

**Comment by WichtP** [comment]
Hello, I activated the debug mode and I send you the result:
➜  ~ tail -f /config/home-assistant.log
  File "/usr/local/lib/python3.10/site-packages/hamsclient/client.py", line 105, in __get_all_stations
    stationData["altitude"] = line["Altitude station m. s. mer"]
  File "/usr/local/lib/python3.10/site-packages/pandas/core/series.py", line 958, in __getitem__
    return self._get_value(key)
  File "/usr/local/lib/python3.10/site-packages/pandas/core/series.py", line 1069, in _get_value
    loc = self.index.get_loc(label)
  File "/usr/local/lib/python3.10/site-packages/pandas/core/indexes/base.py", line 3623, in get_loc
    raise KeyError(key) from err
KeyError: 'Altitude station m. s. mer'
2022-10-01 07:05:02.373 INFO (MainThread) [homeassistant.components.websocket_api.http.connection] [140181826674336] Connection closed by client. It the same problem than Mr. Rodonile. Can everybody help us to solve this. Best regards. PA

**Comment by Superkikim** [comment]
Duplicate: Check this: 

https://githubrecord.com/thor1872

**Comment by albertogmdp** [comment]
Finally!! It works for me now. Thank you very much!

**Comment by WichtP** [comment]
Hello Superkikim,

I tried this morning to change the line on the client.py and it’s working well. Thank you very mutch for your help. Regards.

Wicht Pierre-Alain
Impasse de la Biorda 1
1632 Riaz

**Comment by lpythoud** [comment]
hello, is it possible to know what you did to fix it please ?

**Comment by Superkikim** [comment]
> 1632 Riaz 

On est pas très loin... Je suis à Siviriez ;-)

**Comment by Superkikim** [comment]
> hello, is it possible to know what you did to fix it please ?

Check above. I shared a link. 

That said, I didn't got the error anymore but I couldn't say it worked. 

@WichtP is it fully working for you ? Or were you just able to configure it without error, but then don't get values ? 

In between, I've checked how to get the data myself, in order to perhaps contribute. But I'm in a place where I have no energy and no motivation for that :( 

Here is a link to the python script I made so far.

https://akim.sissaoui.com/informatique/python-download-zip-and-parse-url-from-text-file/

**Comment by lpythoud** [comment]
thanks Superkikim, sorry but impossible to use your link or i don't know how to use it ^^ thanks

**Comment by lpythoud** [comment]
finally, i don't know if this the same issue, when i try to install the integration i have this error message : Le flux de configuration n'a pas pu être chargé: 500 Internal Server Error Server got itself in trouble

Please find below DEBUG log : 
<img width="1506" alt="Capture d’écran 2022-11-11 à 23 37 50" src="https://user-images.githubusercontent.com/10260314/201440260-b2ab6c81-a401-4c1a-a5c0-efc176955b93.png">

Thanks

**Comment by Superkikim** [comment]
Yes it was the same issue. And it seems the post I shared above from thor1872 has been deleted :(  However I have found the original on Github and corrected the link. Here it is: https://github.com/websylv/homeassistant-meteoswiss/issues/65#issuecomment-1199784205

**Comment by lpythoud** [comment]
Thanks @Superkikim  it's working well now !!! have a good day

**Comment by WichtP** [comment]
Bonjour,

J’ai aussi galéré pour trouvé la solution, car le lien n’était plus valide. En fait il faut éditer le fichier client.py comme décrit ci-dessous avec la commande vi

 vi /user/local/lib/python3.10/site-packages/hamsclient/client.py

Pour pouvoir le faire en mode terminal il faut d’abord installer SSH & Web Terminal Add-on puis taper la commande docker exec -it homeassistant /bin/bash

Attention les commandes pour pour l’utilitaire vi sont assez spéciale c’est de l’unix je crois. (Je suis un débutant et je connais pas votre niveau) Vous les trouvez sous ce lien entre autre:

https://linux.goffinet.org/administration/traitement-du-texte/editeur-de-texte-vi/

Si ça peut vous aider. Bonne chance. PA

Meteo-Swiss

To change this in client.py, I had to use the ”SSH and Web terminal” add-on and disable the safe mode. Then type this command : docker exec -it homeassistant /bin/bash And after that : vi /user/local/lib/python3.10/site-packages/hamsclient/client.py to modify the file. Ctrl z z to save and exit. After that only reboot and add the integration again
Just so people know it's "usr" and not "user"
So for the beginner like me this is what you need to do.
1: Install SSH & Web Terminal Add-on
2: Disable the protection mode and enable show in sidebar in the setting. Don't forget to add a password in the config tab of this Add-on, if you don't it won't work.
3: Reboot VM or PI (maybe not necessary) but do it anyway. And start SSH & Web Terminal Add-on after reboot if start on boot is disabled.
4: Click on Terminal in the sidebar
5: Type or past this command: docker exec -it homeassistant /bin/bash
(You can use shift + insert to past it in the terminal.)
6: Type or past this command: vi /usr/local/lib/python3.10/site-packages/hamsclient/client.py
7: Edit this line: stationData["altitude"] = line["Altitude station m. s. mer"]
to this:
stationData["altitude"] = line["Altitude station m s. mer"] (no point after the "m").
8: Shift z z to save and close.
9: Reboot the VM or PI
10: Add integration like you normaly do and enjoy ;)
And don't forget to stop SSH & Web Terminal Add-on and enable the protection mode of the add-on after using it for security reason.
I have tested this and it work.
nano /usr/local/lib/python3.10/site-packages/hamsclient/client.py

Wicht Pierre-Alain
Impasse de la Biorda 1
1632 Riaz

**Comment by Rudd-O** [comment]
I've forked it:

https://github.com/Rudd-O/homeassistant-meteoswiss

Mine works.
